src/ServiceControl.Monitoring.Tests/Infrastructure/EntriesBuilder.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace ServiceControl.Monitoring.Tests.Infrastructure
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Messaging;
+    using NUnit.Framework;
+
+    public static class EntriesBuilder
+    {
+        public static RawMessage.Entry[] Build(Dictionary<DateTime, long> measurements)
```

Comment:
If this wasn't for tests, I'd argue that an array of `{DateTime,long}` tuples would be much better. It's fine for test though.